### PR TITLE
plugin/forward: expose few methods and attributes to public

### PR DESCRIPTION
**Pull request by eugen-kleiner-mesh**
### 1. What does this pull request do?
 * Makes Proxy.Connect() and Forward.List() methods public
 * Adds getters for Proxy.addr, Forward.from and Forward.forceTCP

### 2. Which issues (if any) are related?
It implements https://github.com/coredns/coredns/issues/1762

### CODE REVIEW

**Review by fturib — APPROVED** [review]
Good for me. Thanks!

File: plugin/forward/proxy.go [hunk 1]
```diff
@@ -39,6 +39,9 @@ func NewProxy(addr string, tlsConfig *tls.Config) *Proxy {
 	return p
 }
 
+// Addr returns destination endpoints address to forward to
```
Comment (miekg) [inline plugin/forward/proxy.go]: I would just go with `Addr returns the address to forward to.` (closing dot)

File: plugin/forward/connect.go [hunk 2]
```diff
@@ -33,7 +33,8 @@ func (p *Proxy) updateRtt(newRtt time.Duration) {
 	atomic.AddInt64(&p.avgRtt, int64((newRtt-rtt)/rttCount))
 }
 
-func (p *Proxy) connect(ctx context.Context, state request.Request, forceTCP, metric bool) (*dns.Msg, error) {
+// Connect sets connection and sends the request
```
Comment (miekg) [inline plugin/forward/connect.go]: `Connect selects an upstream, sends the request and waits for a response.` is maybe more descriptive (also closing dot)

**Comment by miekg** [comment]
thanks!

**Pull request merged**